sdk/rust/Cargo.toml

```diff
@@ -15,7 +15,11 @@ tokio = { version = "1.39.2", features = ["rt"] }
 tokio-stream = "0.1.15"
 tonic = "0.12.1"
 schemars = "0.8.21"
+hipcheck-sdk-macros = { path = "../../hipcheck-sdk-macros", version = "0.1.0" }
 
 [build-dependencies]
 anyhow = "1.0.86"
 tonic-build = "0.12.1"
+
+[features]
+macros = []
```

Comment:
this can be updated to conditionally depend on hipcheck-sdk-macros

it's something like:

```rust
[dependencies]
...
hipcheck-sdk-macros = { path = "../../hipcheck-sdk-macros", version = "0.1.0", optional = true }

[features]
macros = ["hipcheck-sdk-macros"]

Reply:
this has the benefit of only compiling the `hipcheck-sdk-macros` crate, if the user specified to use the `macro` crate